mm2src/mm2_core/src/mm_ctx.rs

```diff
@@ -128,6 +132,9 @@ pub struct MmCtx {
     pub db_namespace: DbNamespaceId,
     /// The context belonging to the `nft` mod: `NftCtx`.
     pub nft_ctx: Mutex<Option<Arc<dyn Any + 'static + Send + Sync>>>,
+    /// asynchronous handle for rusqlite connection.
+    #[cfg(not(target_arch = "wasm32"))]
+    pub async_sqlite_connection: Constructible<Arc<AsyncMutex<AsyncConnection>>>,
```

Comment:
Does `AsyncConnection` needs to be wrapped in a mutex? The idea was to have a lock free connection, I tried it locally and it worked with all the tests passing. The only problem we have to look for is data integrity, I think a unit test with multiple async operations that depend on each other can be tried to test this. I think it will be fine since `Message::Execute` takes a mutable ref to the connection https://github.com/KomodoPlatform/komodo-defi-framework/blob/b03715370c3db2a5aaa0b87f58af93d365d1a408/mm2src/db_common/src/async_sql_conn.rs#L254

Reply:
It very depends. We do use locks for db in nfts and on some other places. Also, on current design, without lock we would end up having race condition on concurrent executions. I have to review all the db implementations and then say whether we should avoid doing locks on connection(by offering better design) or not.

Reply:
But iirc for some of the parts in mm2 we could use db transactions instead of locking the connection.

Reply:
+ having the lock on program level can be by passed when/if we run multiple mm2 instances on the same machine. For that cases it's safer to do locks on db level instead of program(mm2) level. See https://www.sqlite.org/lockingv3.html

Reply:
In current nft impl we have two locks actually.
```rust
let _lock = nft_ctx.guard.lock().await;

let storage = nft_ctx.lock_db().await?;
```
`Async connection` without mutex should work fine in nft rpcs due to `guard.lock()`. 
But yeah I decided to leave async mutex for now to avoid race condition.

Reply:
```rs
let _lock = nft_ctx.guard.lock().await;

let storage = nft_ctx.lock_db().await?;
```

This looks quite wrong. If we have lock on db connection, we shouldn't need these. I think I mention this problem earlier (having multiple locks back-and-forth).

Reply:
I strongly believe mm2 db implementations(including nft) has performance impact by at very least -%10 on the runtime.

Reply:
> This looks quite wrong. If we have lock on db connection, we shouldn't need these. I think I mention this problem earlier (having multiple locks back-and-forth).

Yeah, you are right, works fine without guard. Will fix tomorrow with other notes.

Reply:
Removed `guard` from `NftCtx`

Reply:
Thanks for the discussion, we have to keep the lock for now for simplicity. It will help us avoid other guards/locks. I still think all the current database operations will work without the lock fine except NFT where it will either need a guard or db lock.